src/models/post/stream.rs

```diff
@@ -57,36 +40,37 @@ impl PostStream {
     }
 
     pub async fn get_posts(
-        post_stream_values: PostStreamValues,
-        post_stream_filters: PostStreamFilters,
+        stream_params: PostStreamQuery,
     ) -> Result<Option<Self>, Box<dyn Error + Send + Sync>> {
         // Decide whether to use index or fallback to graph query
         let use_index = Self::can_use_index(
-            &post_stream_filters.sorting,
-            &post_stream_values.author_id,
-            &post_stream_filters.source,
-            &post_stream_values.tags,
+            stream_params.sorting.as_ref().unwrap(),
```

Comment:
Clunky. Easier to solve if we pass just the right few components to `get_post()`